recognition/include/pcl/recognition/mask_map.h

```diff
@@ -65,10 +65,16 @@ namespace pcl
       inline const unsigned char* 
       getData () const { return (&data_[0]); }
 
+      [[deprecated("output parameter is not desirable,\
+              please use newer version")]]
       static void
       getDifferenceMask (const MaskMap & mask0,
                          const MaskMap & mask1,
                          MaskMap & diff_mask);
+      //[[nodiscard]] waiting for C++17
```

Comment:
What about adding `PCL_NODISCARD`, similar to [`PCL_FALLTHROUGH`](https://github.com/PointCloudLibrary/pcl/blob/96b6f4ef177c5ddef8f76492752d55577edb21e6/common/include/pcl/pcl_macros.h#L353-L365)?

Reply:
Sorry, but I can't see where PCL_FALLTHROUGH is defined

Reply:
Sorry, but I can't see where PCL_FALLTHROUGH is defined

Reply:
There's a file in common module `pcl_macros.h`

Reply:
https://github.com/PointCloudLibrary/pcl/blob/96b6f4ef177c5ddef8f76492752d55577edb21e6/common/include/pcl/pcl_macros.h#L353-L365